src/core/utils/platform.ts

```diff
@@ -114,3 +224,220 @@ export function convertPlatformAssetToAssetApiResponse(
     transferable: asset.transferable,
   } as AssetApiResponse;
 }
+
+const convertPlatformTransactionAssetToAssetApiResponse = (
+  asset: PlatformTransactionAsset | undefined,
+): AssetApiResponse | undefined => {
+  if (!asset) return undefined;
+
+  const chainId = Number(asset.chainId);
+  const parsedChainId = Number.isNaN(chainId) ? 0 : chainId;
+
+  const networks = convertTransactionNetworks(asset.networks);
+  const bridgingNetworks = convertTransactionBridgingNetworks(
+    asset.bridging?.networks,
+  );
+
+  const bridgeable =
+    asset.bridging?.bridgeable ??
+    Object.values(bridgingNetworks).some((network) => network.bridgeable);
+
+  const price = asset.price
+    ? {
+        value: toNumber(asset.price.value),
+        changed_at: normalizeTimestamp(asset.price.changedAt),
+        relative_change_24h: toNumber(asset.price.relativeChange24h),
+      }
+    : undefined;
+
+  const colors = convertTransactionColors(asset.colors);
+  const transferable =
+    asset.hasTransferable === true ? asset.transferable : undefined;
+
+  return {
+    asset_code: asset.assetCode as AddressOrEth,
+    chain_id: parsedChainId,
+    name: asset.name,
+    symbol: asset.symbol,
+    decimals: asset.decimals,
+    icon_url: asset.iconUrl,
+    type: asset.type as AssetApiResponse['type'],
+    network: asset.network as ChainName | undefined,
+    colors,
+    price,
+    networks: networks as AssetApiResponse['networks'],
```

Comment:
This and `bridging.networks` could be a bit dangerous too; would use the network helper utils here as well